yieldprotocol/governance-v2
# Lower dust levels to ~$1000

# Proposal
Lower the dust levels

# Background
The dust (min debt) level for USDC and DAI borrowing is set to 5000 (USD VALUE) currently, which now too high to incentivize smaller vaults. Liquidating a vault costs $20 USD at 13 Gwei.

# Details
The [updateDust.ts](https://github.com/yieldprotocol/environments-v2/blob/3b3425137007ea5c81e7466099ea3b187db3343c/scripts/governance/updateDust/updateDust.ts) script will be used, with this input:
```
/// @notice Minimum debt limits
/// @param base identifier (bytes6 tag)
/// @param collateral identifier (bytes6 tag)
/// @param minimum vault debt, modified by decimals
export const newDebtMin: Array<[string, string, number]> = [
  [DAI, ETH, 1000],     // Value in DAI
  [DAI, USDC, 1000],    // Value in DAI
  [DAI, WBTC, 1000],    // Value in DAI
  [DAI, LINK, 1000],    // Value in DAI
  [DAI, FRAX, 1000],    // Value in DAI
  [DAI, UNI, 1000],     // Value in DAI
  [DAI, WSTETH, 1000],  // Value in DAI
  [DAI, ENS, 1000],     // Value in DAI
  [USDC, ETH, 1000],    // Value in USDC
  [USDC, DAI, 1000],    // Value in USDC
  [USDC, WBTC, 1000],   // Value in USDC
  [USDC, LINK, 1000],   // Value in USDC
  [USDC, FRAX, 1000],   // Value in USDC
  [USDC, UNI, 1000],    // Value in USDC
  [USDC, WSTETH, 1000], // Value in USDC
  [USDC, ENS, 1000],    // Value in USDC
  [ETH, USDC, 1],       // Value in ETH
  [ETH, DAI, 1],        // Value in ETH
  [ETH, WBTC, 1],       // Value in ETH
  [ETH, LINK, 1],       // Value in ETH
  [ETH, FRAX, 1],       // Value in ETH
  [ETH, UNI, 1],        // Value in ETH
  [ETH, WSTETH, 1],     // Value in ETH
  [ETH, ENS, 1],        // Value in ETH
]

/// @notice Minimum debt limits to be used in an auction
/// @param collateral identifier (bytes6 tag)
/// @param Minimum debt, modified by decimals
export const newAuctionMin: Array<[string, number]> = [
  [ETH, 1000000],    // Value in terawei (ETH*10**-6) 
  [DAI, 1000],       // Value in DAI
  [USDC, 1000],      // Value in USDC
  [WBTC, 500],       // Value in WBTC basis points (WBTC*10**-4)
  [WSTETH, 1000000], // Value in terawei (ETH*10**-6)
  [LINK, 138],       // Value in LINK
  [ENS, 64],         // Value in ENS
  [UNI, 153],        // Value in UNI
  [FRAX, 1000],      // Value in FRAX
]
```
# Testing
The change has been tested on a mainnet fork and a Tenderly fork by running [updateDust.ts](https://github.com/yieldprotocol/environments-v2/blob/3b3425137007ea5c81e7466099ea3b187db3343c/scripts/governance/updateDust/updateDust.ts) first, followed by [updateDust.test.ts](https://github.com/yieldprotocol/environments-v2/blob/3b3425137007ea5c81e7466099ea3b187db3343c/scripts/governance/updateDust/updateDust.test.ts) to verify that the changes were made.
```
 ~/dev/yield/environments-v2   increase-dust ● ? ⍟12  bash scripts/governance/update/updateDust/updateDust.sh
++ dirname scripts/governance/update/updateDust/updateDust.sh
+ HERE=scripts/governance/update/updateDust
+ export CONF=/Users/devtooligan/dev/yield/environments-v2/scripts/governance/update/updateDust/updateDust.config
+ CONF=/Users/devtooligan/dev/yield/environments-v2/scripts/governance/update/updateDust/updateDust.config
+ RUN='npx hardhat run --network tenderly'
+ npx hardhat run --network tenderly scripts/governance/update/updateDust/updateDust.test.ts
Impersonating 0xfe90d993367bc93D171A5ED88ab460759DE2bED6 on Tenderly

Post deployment state testing
=============================

Debt Limits (dust)
Success! DAI/ETH set to: 1000
Success! DAI/USDC set to: 1000
Success! DAI/WBTC set to: 1000
Success! DAI/LINK set to: 1000
Success! DAI/FRAX set to: 1000
Success! DAI/UNI set to: 1000
Success! DAI/WSTETH set to: 1000
Success! DAI/ENS set to: 1000
Success! USDC/ETH set to: 1000
Success! USDC/DAI set to: 1000
Success! USDC/WBTC set to: 1000
Success! USDC/LINK set to: 1000
Success! USDC/FRAX set to: 1000
Success! USDC/UNI set to: 1000
Success! USDC/WSTETH set to: 1000
Success! USDC/ENS set to: 1000
Success! ETH/USDC set to: 1
Success! ETH/DAI set to: 1
Success! ETH/WBTC set to: 1
Success! ETH/LINK set to: 1
Success! ETH/FRAX set to: 1
Success! ETH/UNI set to: 1
Success! ETH/WSTETH set to: 1
Success! ETH/ENS set to: 1

Auction Limits (dust)
Success! ETH set to: 1000000
Success! DAI set to: 1000
Success! USDC set to: 1000
Success! WBTC set to: 500
Success! WSTETH set to: 1000000
Success! LINK set to: 138
Success! ENS set to: 64
Success! UNI set to: 153
Success! FRAX set to: 1000
```
